notes/change-report-form-received.vbs

```diff
@@ -65,55 +70,72 @@ BeginDialog Dialog1, 0, 0, 376, 280, "Change Report Form Received"
   Text 210, 10, 110, 10, "Date Change Reported/Received:"
   GroupBox 5, 25, 365, 150, "Changes Reported:"
   Text 15, 40, 30, 10, "Address:"
   Text 15, 60, 35, 10, "HH Comp:"
-  Text 15, 80, 95, 10, "Assets (savings or property):"
-  Text 15, 100, 30, 10, "Vehicles:"
-  Text 15, 120, 30, 10, "Income:"
-  Text 15, 140, 25, 10, "Shelter:"
+  Text 15, 80, 30, 10, "Income:"
+  Text 15, 100, 25, 10, "Shelter:"
+  Text 15, 120, 95, 10, "Assets (savings or property):"
+  Text 15, 140, 30, 10, "Vehicles:"
   Text 15, 160, 20, 10, "Other:"
   Text 10, 185, 45, 10, "Action Taken:"
   Text 10, 205, 45, 10, "Other Notes:"
   Text 10, 225, 60, 10, "Verifs Requested:"
   Text 10, 265, 60, 10, "Worker Signature:"
   Text 180, 245, 90, 10, "The changes client reports:"
 EndDialog
-'Shows dialog
+
+
+' Updated dialog to include field validation
 DO
 	DO
-		DO
-			DO
-    			Dialog Dialog1
-				cancel_confirmation
-				IF worker_signature = "" THEN MsgBox "You must sign your case note!"
-			LOOP UNTIL worker_signature <> ""
-			IF IsNumeric(MAXIS_case_number) = FALSE THEN MsgBox "You must type a valid numeric case number."
-		LOOP UNTIL IsNumeric(MAXIS_case_number) = TRUE
-		IF changes_continue = "Select One:" THEN MsgBox "You Must Select 'The changes client reports field'"
-	LOOP UNTIL changes_continue <> "Select One:"
-	call check_for_password(are_we_passworded_out)  'Adding functionality for MAXIS v.6 Passworded Out issue'
+		err_msg = ""
+		Dialog Dialog1
+		cancel_confirmation ' Asks user to confirm if they want to cancel
+
+		' Validate that MAXIS number is numeric and less than 8 digits long
+		CALL validate_MAXIS_case_number(err_msg, "* ")
+		' Validate that MAXIS case number is valid
+		CALL navigate_to_MAXIS_screen("CASE", "NOTE")
+		EMReadScreen invalid_case_number, 7, 24, 2
+		Call back_to_SELF()
+		If invalid_case_number = "INVALID" THEN err_msg = err_msg & vbNewLine & "* This is an invalid case number - update the case number."
+		' Validate that worker signature is not blank.
+		IF trim(worker_signature) = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
+		' Validate that worker selects option from dropdown list as to how long change will last
+		If changes_continue = "Select One:" THEN err_msg = err_msg & vbNewLine & "* You must select an option from the dropdown list indicating whether the changes reported by the client will continue next month or will not continue next month."
+		' Validate the fields to ensure that at least one field is filled in besides the required fields (case number, worker signature, select one dropdown)
+		If date_received = "" AND effective_date = "" AND address_notes = "" AND household_notes = "" AND asset_notes = "" AND vehicles_notes = "" AND income_notes = "" AND shelter_notes = "" AND other_change_notes = "" AND actions_taken = "" AND other_notes = "" AND verifs_requested = "" THEN err_msg = err_msg & vbNewLine & "* All of the fields are blank. You must enter information in at least one field."
```

Comment:
date_received and effective_date: I would create separate if statements for these and and add date handling. I was able to enter a string a random characters vs. being able to enter a date.

Reply:
I added in date handling to ensure that the entry was in a 10-character date format and could not be blank